sklearn/utils/estimator_checks.py

```diff
@@ -1190,6 +1190,30 @@ def check_estimators_pickle(name, estimator_orig):
         unpickled_result = getattr(unpickled_estimator, method)(X)
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
+    # check if input with nans pickles properly
+    if name in ALLOW_NAN:
+        # set random 10 elements to be corrupted
+        X.ravel()[np.random.choice(X.size, 10, replace=False)] = np.nan
+        estimator.fit(X, y)
+
+        nan_result = dict()
+        for method in check_methods:
+            if hasattr(estimator, method):
+                nan_result[method] = getattr(estimator, method)(X)
+
+        # pickle and unpickle!
+        pickled_estimator = pickle.dumps(estimator)
+        if estimator.__module__.startswith('sklearn.'):
+            assert_true(b"version" in pickled_estimator)
```

Comment:
That sounds unrelated to detecting nans. Why is it necessary?

Reply:
yep, I removed the unnecessary portion